### PR TITLE
[reflection] Implement RuntimeType.IsSubclassOf in native

### PR DESCRIPTION
**Pull request by lambdageek**
On my machine the benchmark in #10000 takes `889` ms.   Using this native version, I get `222`ms.  So approximately a 4x speedup.

### CODE REVIEW

**Comment by lambdageek** [comment]
@monojenkins build pkg

**Comment by lambdageek** [comment]
@monojenkins backport to 2018-08

**Comment by lambdageek** [comment]
@monojenkins backport to 2018-06

File: mcs/class/corlib/ReferenceSources/RuntimeType.cs [hunk 1]
```diff
@@ -812,5 +812,20 @@ public override int GetHashCode()
...
+		{
+			if ((object)type == null)
+				throw new ArgumentNullException("type");
+			Contract.EndContractBlock();
+			RuntimeType rtType = type as RuntimeType;
+			if (rtType == null)
+				return false;
+
+			return RuntimeTypeHandle.IsSubclassOf (this, rtType);
+		}
```
Comment (akoeplinger) [inline mcs/class/corlib/ReferenceSources/RuntimeType.cs]: 90% of this method is the same as the referencesource version so I'd keep this there and just put the `RuntimeTypeHandle.IsSubclassOf ` in the ifdef.

(probably not worth going through another CI iteration cycle for this, unless you need to make changes anyway)

**Comment by jaykrell** [comment]
Corlib version?

**Pull request merged**